src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1632,26 +1632,28 @@ private bool IsPartialType()
             return false;
         }
 
-        private bool IsPartialMember()
+        private bool IsPartialMember(bool allowPartialCtor)
```

Comment:
my preference is to not have ambient state.  that makes parsing (esp. incremental parsing) more difficult.  Is it possible to just answer this uniformly regardless on ambient state?  It might mean parsing something out which needs a binding error later, but that's fine.

Reply:
I don't like that either, but I need to avoid breaks in cases like this:

```cs
System.Console.Write(F().GetType().Name);
partial F() => new(); // don't want to parse this as a constructor in top-level statements
class @partial;
```